drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,482 @@
+int max9611_read(struct max9611_dev *dev, uint8_t addr,
+		 uint8_t bytes, uint8_t *read_data)
+{
+	int ret;
+
+	/* Check if bytes is 0 */
+	if (bytes <= 0)
+		return -EINVAL;
+
+	/* Check if valid register */
+	if (addr < MAX9611_CSA_MSB || addr > MAX9611_CTR2)
+		return -EINVAL;
+
+	/* Avoid read to out of bounds register */
+	if (addr + bytes > MAX9611_OUT_OF_BOUNDS)
+		return -EINVAL;
+
+	ret = no_os_i2c_write(dev->i2c_desc, &addr, 1, 0);
+	if (ret)
+		return ret;
+
+	return no_os_i2c_read(dev->i2c_desc, read_data, bytes, 1);
+}
+
+/***************************************************************************//**
+ * @brief Writes data to device register (I2C)
+ * @param dev 		  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param bytes 	  - Number of bytes to write from starting register address
+ * @param write_data  - Write data buffer
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_write(struct max9611_dev *dev, uint8_t addr,
+		  uint8_t bytes, uint8_t *write_data)
+{
+	int i;
+	uint8_t buf[bytes + 1];
+
+	/* Check if bytes is 0 */
+	if (bytes <= 0)
+		return -EINVAL;
+
+	/* Check if valid register */
+	if (addr < MAX9611_CSA_MSB || addr > MAX9611_CTR2)
+		return -EINVAL;
+
+	/* Avoid write to out of bounds register */
+	if (addr + bytes > MAX9611_OUT_OF_BOUNDS)
+		return -EINVAL;
+
+	/* Exit when writing to Read only registers */
+	if (addr < MAX9611_CTR1)
+		return -EINVAL;
+
+	buf[0] = addr;
+
+	for (i = 0; i < bytes; i++)
+		buf[i + 1] = write_data[i];
+
+	return no_os_i2c_write(dev->i2c_desc, buf, bytes + 1, 1);
+}
+
+/***************************************************************************//**
+ * @brief Update selected I2C register contents
+ * @param dev   	  - The device structure.
+ * @param addr 		  - Starting / selected register address.
+ * @param update_val  - Value used to update on selected register
+ * @param update_mask - Mask to update to selected register bit/s
+ * @return            - 0 if I2C write is successful, error otherwise.
+*******************************************************************************/
+int max9611_reg_update(struct max9611_dev *dev, uint8_t addr,
+		       int update_val, int update_mask)
+{
+	uint8_t temp;
+	int ret;
+
+	ret = max9611_read(dev, addr, 1, &temp);
+	if (ret)
+		return ret;
+
+	temp &= ~update_mask;
+	temp |= update_val;
+
+	return max9611_write(dev, addr, 1, &temp);
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 operating mode
+ * @param dev 		  - The device structure.
+ * @param mode 		  - MODE configuration
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_mode(struct max9611_dev *dev, enum max9611_mode_conf mode)
```

Comment:
you can call `no_os_field_prep` inline:
`max9611_reg_update(dev, MAX9611_CTR1, no_os_field_prep(MAX9611_MODE_MASK, mode),
				  MAX9611_MODE_MASK);`

Reply:
Implemented in latest commit.